### PR TITLE
Cleaned up Kotlin android example

### PR DESCRIPTION
**Pull request by zhaonian**

### CODE REVIEW

File: examples/example-kotlin/android/helloworld/app/src/main/res/layout/activity_helloworld.xml [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
+        android:layout_height="match_parent"
+        tools:context=".MainActivity"
+        android:orientation="vertical" >
 
     <LinearLayout
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:orientation="horizontal">
         <EditText
-                android:id="@+id/host_edit_text"
```
Comment (voidzcy) [inline examples/example-kotlin/android/helloworld/app/src/main/res/layout/activity_helloworld.xml]: We do not have a strictly defined guideline for resource ID naming, and believe the author's convention is "IDs suffixed with the name of the element in lowercase underscore". I found the most commonly used convention is to ["prefix with the name of the element in lowercase underscore"](https://stackoverflow.com/questions/35923922/best-practices-for-id-naming-conventions-in-android). This is an opinion-based answer. But, we would prefer name resource IDs in lowercase underscore, and we used the same convention in other Android examples in grpc-java repo.

Comment (zhaonian) [reply examples/example-kotlin/android/helloworld/app/src/main/res/layout/activity_helloworld.xml]: Done.
Thx for the review!

File: examples/example-kotlin/android/helloworld/app/src/main/kotlin/io/grpc/helloworldexample/HelloworldActivity.kt [hunk 2]
```diff
@@ -31,33 +33,38 @@ import io.grpc.ManagedChannel
...
+import kotlinx.android.synthetic.main.activity_helloworld.grpc_response_text
+import kotlinx.android.synthetic.main.activity_helloworld.send_button
+import kotlinx.android.synthetic.main.activity_helloworld.host_edit_text
+import kotlinx.android.synthetic.main.activity_helloworld.message_edit_text
+import kotlinx.android.synthetic.main.activity_helloworld.port_edit_text
 import java.io.PrintWriter
 import java.io.StringWriter
 import java.lang.ref.WeakReference
 import java.util.concurrent.TimeUnit
-import kotlinx.android.synthetic.main.activity_helloworld.*
```
Comment (voidzcy) [inline examples/example-kotlin/android/helloworld/app/src/main/kotlin/io/grpc/helloworldexample/HelloworldActivity.kt]: This is an excessive change, `kotlinx.android.synthetic.main.activity_helloworld.*` is better.

File: examples/example-kotlin/android/helloworld/app/src/main/kotlin/io/grpc/helloworldexample/HelloworldActivity.kt [hunk 3]
```diff
@@ -31,33 +33,38 @@ import io.grpc.ManagedChannel
...
-    send_button!!.setOnClickListener(this)
+    grpc_response_text?.movementMethod = ScrollingMovementMethod()
+    send_button?.setOnClickListener(this)
   }
 
+  @RequiresApi(Build.VERSION_CODES.CUPCAKE)
   override fun onClick(view: View) {
     (getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager)
-        .hideSoftInputFromWindow(host_edit_text!!.windowToken, 0)
-    send_button!!.isEnabled = false
```
Comment (voidzcy) [inline examples/example-kotlin/android/helloworld/app/src/main/kotlin/io/grpc/helloworldexample/HelloworldActivity.kt]: Ditto.

File: examples/example-kotlin/android/helloworld/app/src/main/kotlin/io/grpc/helloworldexample/HelloworldActivity.kt [hunk 4]
```diff
@@ -31,33 +33,38 @@ import io.grpc.ManagedChannel
...
-    grpc_response_text!!.text = ""
+        .hideSoftInputFromWindow(host_edit_text?.windowToken, 0)
+    send_button?.isEnabled = false
+    grpc_response_text?.text = ""
     GrpcTask(this)
         .execute(
-            host_edit_text!!.text.toString(),
```
Comment (voidzcy) [inline examples/example-kotlin/android/helloworld/app/src/main/kotlin/io/grpc/helloworldexample/HelloworldActivity.kt]: Ditto.

File: examples/example-kotlin/android/helloworld/app/src/main/kotlin/io/grpc/helloworldexample/HelloworldActivity.kt [hunk 5]
```diff
@@ -31,33 +33,38 @@ import io.grpc.ManagedChannel
...
 import java.util.concurrent.TimeUnit
-import kotlinx.android.synthetic.main.activity_helloworld.*
 
 class HelloworldActivity : AppCompatActivity(), View.OnClickListener {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_helloworld)
-    grpc_response_text!!.movementMethod = ScrollingMovementMethod()
-    send_button!!.setOnClickListener(this)
+    grpc_response_text?.movementMethod = ScrollingMovementMethod()
```
Comment (voidzcy) [inline examples/example-kotlin/android/helloworld/app/src/main/kotlin/io/grpc/helloworldexample/HelloworldActivity.kt]: We would rather have exception thrown sooner than got into some weird unexpected state.

**Comment by voidzcy** [comment]
> We do not have a strictly defined guideline for resource ID naming, and believe the author's convention is "IDs suffixed with the name of the element in lowercase underscore". I found the most commonly used convention is to ["prefix with the name of the element in lowercase underscore"](https://stackoverflow.com/questions/35923922/best-practices-for-id-naming-conventions-in-android). This is an opinion-based answer. But, we would prefer name resource IDs in lowercase underscore, and we used the same convention in other Android examples in grpc-java repo.


Update: I just realized that the resource ID naming convention may not apply for Kotlin Android Extensions (intended for the traditional `findViewById(...)`). It might look more natural with camel case naming as they are directly imported via Kotlin Android Extensions. But I do not find an official style guide for that and this seems to be an [controversial topic](https://www.reddit.com/r/androiddev/comments/30vy3y/how_do_you_name_your_view_ids/). I looked up some Kotlin Android apps in Google's internal codebase, they are following the underscore convention for resource IDs, so we would just follow this.

**Comment by zhaonian** [comment]
I see.
I guess I'll just close this PR for now because my goal was avoiding !!
Thanks for the review! :) @voidzcy

**Pull request closed without merging**